src/Workspaces/CSharp/Portable/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -40,6 +40,13 @@ internal static partial class ITypeSymbolExtensions
         private static TypeSyntax GenerateTypeSyntax(
             INamespaceOrTypeSymbol symbol, bool nameSyntax, bool allowVar = true)
         {
+            if (symbol is ITypeSymbol type && type.ContainsAnonymousType())
```

Comment:
Do we have a unit test for this case?

Reply:
Yes.  this is covered already by tests for introduce-local.